### PR TITLE
Cypress Test fixes for event-analytics

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
Fixed the re-naming of event-analytics links causing the testing to fail.
Fixed existing bugs in some of the test so each run successfully 

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .cypress/integration/1_event_analytics.spec.js [hunk 1]
```diff
@@ -45,12 +45,11 @@ describe('Adding sample data and visualization', () => {
...
   it('Redirect to correct page on breadcrumb click', () => {
     landOnEventExplorer();
-    cy.get('.euiBreadcrumb[href="#/event_analytics/explorer"]').contains('Explorer').click();
-    cy.get('[data-test-subj="searchAutocompleteTextArea"]').should('exist');
-    cy.get('.euiBreadcrumb[href="#/event_analytics"]').contains('Event analytics').click();
-    cy.get('.euiTitle').contains('Event analytics').should('exist');
-    cy.get('.euiBreadcrumb[href="observability-dashboards#/"]').contains('Observability').click();
-    cy.get('.euiTitle').contains('Event analytics').should('exist');
+    suppressResizeObserverIssue();
+    cy.wait(delay);
```
Comment (mengweieric) [inline .cypress/integration/1_event_analytics.spec.js:49]: We want to avoid any explicit wait as it is considered as anti-pattern which may lead to flaky or inefficient tests. You may add timeout to below 'get' method to be something like ```cy.get('.euiBreadcrumb[href="observability-logs#/"]', {timeout: xxx}).click();```it will retry and wait for the element to appear in DOM.

File: .cypress/integration/1_event_analytics.spec.js [hunk 2]
```diff
@@ -214,103 +215,16 @@ describe('Click actions', () => {
...
-    cy.url().should('contain', '#/event_analytics/explorer');
-    cy.get('[data-test-subj="searchAutocompleteTextArea"]', {
-      timeout: COMMAND_TIMEOUT_LONG,
-    }).contains(TEST_QUERIES[0].query);
-  });
-});
-
-describe('Override timestamp for an index', () => {
+//  Waiting for PR451
+describe.skip('Override timestamp for an index', () => {
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply .cypress/integration/1_event_analytics.spec.js]: Done, removed the skip

File: .cypress/integration/1_event_analytics.spec.js [hunk 3]
```diff
@@ -545,3 +463,93 @@ describe('Visualizing data', () => {
     cy.get('@legandTxt').should('contain', BAR_LEG_TEXT_3);
   });
 });
+
+describe('Saves a query on explorer page', () => {
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply .cypress/integration/1_event_analytics.spec.js:466]: Order was impacting an error occurring on cypress's side labeled as a "node_modules/async-wait-until/src/waitUntil.js:53:1" putting this test at the end / commenting it out allowed the rest of the test to run successfully.

File: package.json [hunk 4]
```diff
@@ -44,7 +44,7 @@
     "@types/react-plotly.js": "^2.5.0",
     "@types/react-test-renderer": "^16.9.1",
     "antlr4ts-cli": "^0.5.0-alpha.4",
-    "cypress": "^6.0.0",
+    "cypress": "12.11.0",
```
Comment (mengweieric) [inline package.json:47]: Maybe better to upgrade it in a separate PR after we have all-green status for all the tests

File: .cypress/integration/1_event_analytics.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mengweieric) [reply .cypress/integration/1_event_analytics.spec.js:462]: We may want to know the exact error here to fix it without changing the order.

[the remark this answers is not in the document]
Comment (pjfitzgibbons) [reply .cypress/integration/1_event_analytics.spec.js:462]: I want to know why order should matter?

TLDR; The test suite should run in any order. This is imperative.
IF there is a test that is order-dependent, then it is an erroneous test and is high/critical on the list of things-to-fix.

In the course of debugging, this describe() was definitively causing the test-suite to fail.
Moving this test simplified debugging and will simplify fixing it in a subsequent PR.

File: package.json (hunk 4 above)
Comment (pjfitzgibbons) [reply package.json:47]: There are some test failures that dont' seem to be fixable in Cypress 6.
Cypress has had significant refactors (6 Majors!).  The internal browser-automation has been changed. (Investigating 6 majors worth the code-changes is infeasible).
Some browser-automation techinques from Cypress 6 -> 12.x have changed in ways that are difficult to document.  The APIs have changed, which provides a small clue.  Other changes are simply observed and trial-and-error + intuition is necessary to determine the "correct" repeatable automation.

File: cypress.json
[the remark this answers is not in the document]
Comment (TackAdam) [reply cypress.json:1]: Has to be removed in order to run Cypress version 12, replaced with cypress.config.js (Pending commit depending which version we end up using)

**Comment by TackAdam** [comment]
Removed version upgrade as it will be a separate PR pending discussion on https://github.com/opensearch-project/dashboards-observability/issues/411.
This PR now just fixes the url naming conventions and wait commands.

**Review by derek-ho — APPROVED** [review]
🚀

**Review by kavithacm — APPROVED** [review]
LGTM!

**Review by pjfitzgibbons — APPROVED** [review]
